Source/NSubstitute/Routing/RouteFactory.cs

```diff
@@ -57,6 +62,7 @@ public IRoute RecordCallSpecification(ISubstituteState state)
         {
             return new Route(new ICallHandler[] {
                 new RecordCallSpecificationHandler(state.PendingSpecification, state.CallSpecificationFactory, state.CallActions)
+                , new ClearLastTrackedCallHandler(state.LastCallTracker)
```

Comment:
I'm a bit confused by this line. Why do you clear last call in `RecordCallSpecification`?